libbpf-rs/src/object.rs

```diff
@@ -361,9 +361,11 @@ impl Object {
             let section = util::c_ptr_to_string(section)?;
 
             // Get the program fd
-            let fd = unsafe { libbpf_sys::bpf_program__fd(next_ptr) };
-            if fd < 0 {
-                return Err(Error::System(-fd));
+            if unsafe { libbpf_sys::bpf_program__autoload(next_ptr) } {
+                let fd = unsafe { libbpf_sys::bpf_program__fd(next_ptr) };
+                if fd < 0 {
+                    return Err(Error::System(-fd));
+                }
```

Comment:
Looking at this some more, I think we can just remove the `libbpf_sys::bpf_program__fd` call altogether - we don't do anything with the program fd. WDYT?

Reply:
Yes, that is what I had in mind. Let's drop it then.